### PR TITLE
Convert project ID to string. Fixes #2170

### LINKED ISSUE #2170: Ensure project ID is string (for easier comparisons)
`project.getId()` returns an ObjectID (from MongoDB) currently. This can make comparisons tricky since we are sometimes comparing with a string and sometimes both are ObjectIDs/strings. It would be easier to simply ensure that the `projectId`s are always strings.

### PR DESCRIPTION
**Pull request by brollb**

### CODE REVIEW

**Comment by hamidzr** [comment]
failing test on travis seems unrelated and due to a slow container 
> 
  1) collaboration requesting actions should be able to request missing actions:
     Error: Timeout of 5000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

File: src/server/storage/projects.js [hunk 1]
```diff
@@ -63,7 +63,7 @@
         }
 
         getId() {
-            return this._id;
+            return this._id.toString();
```
Comment (hamidzr) [inline src/server/storage/projects.js]: are we sure that this ID isn't being used somewhere else directly relying on the fact that this is a valid `ObjectId`

**Comment by brollb** [comment]
Darn. I pulled this change from the other PR I am working on and must have missed a portion there (these were passing when I was working on that branch). I will check it out - good catch :)

**Comment by brollb** [comment]
Fixed the failing tests and made sure the `projectId` fields are strings and not `ObjectId`s. However, we should probably add a migration script for this to migrate the recorded project actions in the database.

**Comment by hamidzr** [comment]
the codeclimate issue is related to es8 and will be fixed by #2175 thanks

**Pull request merged**